solr/core/src/java/org/apache/solr/core/backup/BackupManager.java

```diff
@@ -287,6 +293,51 @@ public void downloadCollectionProperties(String collectionName) throws IOExcepti
     }
   }
 
+  private void downloadFromZK(ConfigSetService configSetService, String configName, URI dir) throws IOException {
+    List<String> files = configSetService.listFilesInConfig(configName);
```

Comment:
I'm guessing listFilesInConfig does only one level?  I see you have it overloaded that takes "file".  This feels a bit awkward to me... can't we just list all files in the configSet?  The code here looks like it might not work if there are multiple intermediate levels (since there is no recursion), but I could be wrong.